js/filters.js

```diff
@@ -152,6 +214,49 @@ DateFilter.prototype.setValue = function(value) {
   this.$maxDate.val(value[1]).change();
 };
 
+DateFilter.prototype.handleModifyEvent = function(e, opts) {
+  // Sets min and max years based on the transactionPeriod filter
+  if (opts.filterName === this.name) {
+    this.maxYear = parseInt(opts.filterValue);
+    this.minYear = this.maxYear - 1;
+    this.$minDate.val('01-01-' + this.minYear.toString()).change();
+    this.$maxDate.val('12-31-' + this.maxYear.toString()).change();
+    this.validate();
+  }
+};
+
+DateFilter.prototype.showWarning = function() {
+  if (!this.showingWarning) {
+    var warning =
+    '<div class="message message--error message--small">' +
+      'You entered a date that\'s outside the selected transaction period. ' +
+      'Please enter a receipt date from ' +
+      '<strong>' + this.minYear + '-' + this.maxYear + '</strong>' +
+    '</div>';
+    this.$maxDate.after(warning);
+    this.showingWarning = true;
+  }
+
+  this.$body.trigger('filters:validation', [
```

Comment:
I wonder if this makes more sense to move up into the `validate` method itself.  The reason being is that this is an action to be taken as a result of the validation checks and not necessarily a part of the alert/warning display.  By having it in here, it's coupled with the warning display logic and this may need to change in the future, or we may want the `isValid` state change triggered by something else entirely, independent of showing/hiding a warning.

Does this make sense?  What are your thoughts?